### ISSUE TITLE
Is there any plan to release the training code?

### ISSUE THREAD

**Issue by trouble-maker007** [lllyasviel/FramePack#21]
It is really amazing job, I was wonder that any plan to release the training code or the sft code?

**Comment by lllyasviel** [comment]
I will add training codes later. training is very straight forward and simple

**Comment by trouble-maker007** [comment]
@lllyasviel I’m reproducing the training code and I have a question: since during inference the video is split into multiple segments based on its duration and FPS, do you also need to compute the loss over multiple noisy segments during training?

**Comment by wangshankun** [comment]
> [@lllyasviel](https://github.com/lllyasviel) I’m reproducing the training code and I have a question: since during inference the video is split into multiple segments based on its duration and FPS, do you also need to compute the loss over multiple noisy segments during training?

I don't think it's necessary. You can refer to the training code of the CausalVid project. Of course, I've also tried the idea you mentioned, but it didn't seem to work very well.

<img width="1145" alt="Image" src="https://github.com/user-attachments/assets/a292dc89-08d6-4856-afbf-37fd22f4ef18" />

**Comment by yuchenli-sony** [comment]
let us know when will the train code be released

**Comment by zhangsongchn** [comment]
Thank you for your work. When will the training code be open sourced?

**Comment by jh27kim** [comment]
Hi @lllyasviel thank you for your effort in publicly sharing the code. Do you plan to release the training code as well?

**Comment by OrangeSodahub** [comment]
Hi, @lllyasviel could you provide a ddl when you will upload the training codes?

**Comment by OrangeSodahub** [comment]
If the author can provide a tentative timeline before which the full codes/details can be released, we will really appreciate it!

**Comment by OrangeSodahub** [comment]
cc @lllyasviel

**Comment by StoneAgeJL** [comment]
> I will add training codes later. training is very straight forward and simple

Thank you for your contribution. Looking forward to it 🙏

**Comment by OrangeSodahub** [comment]
I really want to know an approximate timeline when the full codes can be released... :sob::sob::sob::sob::sob::sob::broken_heart::broken_heart::broken_heart::broken_heart::broken_heart:

**Comment by nankepan** [comment]
> I will add training codes later. training is very straight forward and simple

Months have pass... I really want to know an approximate timeline when the full codes can be released... 😭😭😭😭😭😭💔💔💔💔💔

**Comment by Aurora-edu** [comment]
Thank you for your contribution. Looking forward to it ！When will the training code be released 😭😭😭💔💔💔💔💔

**Comment by Ry3nG** [comment]
> I will add training codes later. training is very straight forward and simple

It's been almost half a year 😭

**Comment by OrangeSodahub** [comment]
I assume that the codes will not be opensoured recently since FramePack seems to be used for some commerical product 😭😭😭💔💔💔💔💔
